[Kelvin Chua] iP

chuakid opened the pull request:
### New PR
> Life before death. Strength before weakness. Journey before destination. - **Brandon Sanderson, [The Way Of Kings](https://www.goodreads.com/book/show/7235533-the-way-of-kings)**
- added new features
- used for learning

Things you can do with it:
1. Fork it
2. Make todos

Features:
- [X] Todos
- [X] Deadline todos
- [X] Event todos

Fun command: `rm -rf /`
Fun hyperlink: [my website](http://kelvinchua.dev)
Fun emoji: 💀
Python code:
```python
while True:
    study()
```

Commit:
Add Gradle support

chuakid committed:
Duke class: add bye and repeating input capability

chuakid committed:
Duke class: add to list functionality

chuakid committed:
Duke class: list items functionality

chuakid committed:
Add Task class

chuakid committed:
Task class: add toString override

chuakid committed:
Duke class: replaced string array with Task array

chuakid committed:
Duke class: check command by first word instead of entire string

chuakid committed:
Duke class: add mark and unmark functionality

chuakid committed:
Prepare Task for subtypes

chuakid committed:
Allow for Todos

chuakid committed:
Modify command detection and fix todo

chuakid committed:
Duke class: add deadline functionality

chuakid committed:
Duke class: add Event functionality
Event class: created

chuakid committed:
Fix bugs and handle errors

chuakid committed:
Convert array to arraylist

chuakid committed:
Added delete functionality

Update test cases

chuakid committed:
Fix switch-case styling

chuakid committed:
Add exceptions

chuakid committed:
Duke class: Add saving and loading functionality

chuakid committed:
Add TaskCreationException
Change date in tasks to LocalDate

chuakid committed:
Merge branch 'branch-Level-7'

chuakid committed:
Merge branch 'branch-Level-8'

chuakid committed:
Fix exception for todo creation

chuakid committed:
Abstract out storage into its own class

chuakid committed:
Abstract out TaskList

chuakid committed:
Abstract out ui

chuakid committed:
Abstract out parsing

chuakid committed:
Move everything into packages

chuakid committed:
Merge branch 'add-gradle-support'

chuakid committed:
Set up gradle

chuakid committed:
Create tests for Todo

chuakid committed:
Create tests for Deadline

chuakid committed:
Fix creation of save file

chuakid committed:
Add javadoc

chuakid committed:
Add javadoc

chuakid committed:
Add javadoc

chuakid committed:
Add javadoc

chuakid committed:
Fix coding standard

chuakid committed:
Add find feature

chuakid committed:
Add tests for find

chuakid committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/Ui.java

chuakid committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/TaskList.java

chuakid committed:
Refactor storage into a non-static class

chuakid committed:
Change save folder generation

chuakid committed:
Add checkstyle and Fix checkstyle violations

chuakid committed:
Change tasklist to package private

chuakid committed:
Remove wildcard import
Make variable final

chuakid committed:
Rename variable

chuakid committed:
Update checkstyle

chuakid committed:
Add javafx

chuakid committed:
Add GUI

kenzantonius reviewed (commented):
Overall, great job! Code is very neat and follows coding standards. Keep up the good work! :D

sembcorpp commented:
Good job for keeping up on task with the schedule! Code is easy to read and follow :))

pangrwa commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,23 @@
+ * Controller for Duke
+ */
+public class Duke {
+    static final String SAVE_FOLDER = "data/save.txt";
```
Might want to consider changing the variable name. `SAVE_FOLDER` sounds a bit like a boolean type to me. Maybe try `SAVED_PATH` instead?

pangrwa commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,84 @@
+ */
+public class Parser {
+
+    private String listALl(TaskList tl) {
```
I think there there's a typo in the method name! `listAL1` isn't really clear.

chuakid replied:
Thanks for spotting this!

pangrwa commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,47 @@
+     * @param list TaskList to be stored
+     */
+    void store(TaskList list) {
+        boolean ignored = new File(savePath).getParentFile().mkdirs();
```
Maybe try naming it as `isIgnored`?

pangrwa commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,119 @@
+    }
+
+    TaskList find(String desc) {
+        return new TaskList(list
```
I like how you simplified the code! Makes it easier to read.

pangrwa commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,42 @@
+ * Represents an event task
+ */
+public class Event extends Task {
+    private final LocalDate from;
```
Might want to consider renaming `from` and `to` to `startDate` and `endDate` which could be a little clearer.

pangrwa reviewed (commented):
Overall, I your code is great. The use of streams to simplify what would have been large chunk of codes is great! Also, one area you could work on is renaming variables!

chuakid committed:
Add assertion

chuakid committed:
Add command class

Command class will be used for executing commands.

chuakid committed:
Fix style errors

chuakid committed:
Major refactor

Implement commands classes for each class

Redefine execute function in Command class

Write javadoc for misc functions

Alter accessibility for functions

Move command logic away from Parser.java

Create new exceptions

chuakid committed:
Merge pull request #1 from chuakid/branch-A-Assertions

Add assertion

chuakid committed:
Merge branch 'branch-A-CodeQuality'

# Conflicts:
#	src/main/java/duke/ui/MainWindow.java

chuakid committed:
Setup github ci

chuakid committed:
Modified exception handling

chuakid committed:
Refactor task classes and add priority

chuakid committed:
Add ability to set priority

chuakid committed:
Show priority in list

chuakid committed:
Refactor listAll into toString for TaskList.java

chuakid committed:
Save after changing priority

chuakid committed:
Sort by priority when listing tasks

chuakid committed:
Fix test

chuakid committed:
Make getter and setter for priority

chuakid committed:
Change tests

chuakid committed:
Fix style problems

chuakid committed:
Add dialog box and style

chuakid committed:
Make VBox fit the width of container

chuakid committed:
Add padding

chuakid committed:
Fix task adding

chuakid committed:
Add user guide

chuakid committed:
Fix bug with erroneous command parsing
